File: C++/1305_AllElements.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> getAllElements(TreeNode* root1, TreeNode* root2) {
        vector<int> visited;
        queue<TreeNode*> q1;
        queue<TreeNode*> q2;
        if (root1)
            q1.push(root1);
        if (root2)
            q2.push(root2);
        while (!q1.empty()){
            TreeNode* node = q1.front();
            visited.push_back(node->val);
            if (node->left)
                q1.push(node->left);
            if (node->right)
                q1.push(node->right);
            q1.pop();
        }
        while (!q2.empty()){
            TreeNode* node = q2.front();
            visited.push_back(node->val);
            if (node->left)
                q2.push(node->left);
            if (node->right)
                q2.push(node->right);
            q2.pop();
        }
        sort(visited.begin(), visited.end());
        return visited;
    }
};

File: C#/1305_AllElements.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<int> GetAllElements(TreeNode root1, TreeNode root2) {
        List<int> visited = new List<int>();
        Queue<TreeNode> q1 = new Queue<TreeNode>();
        Queue<TreeNode> q2 = new Queue<TreeNode>();
        if (root1 != null)
            q1.Enqueue(root1);
        if (root2 != null)
            q2.Enqueue(root2);
        while (q1.Count != 0){
            TreeNode node = q1.Dequeue();
            visited.Add(node.val);
            if (node.left != null)
                q1.Enqueue(node.left);
            if (node.right != null)
                q1.Enqueue(node.right);
        }
        while (q2.Count != 0){
            TreeNode node = q2.Dequeue();
            visited.Add(node.val);
            if (node.left != null)
                q2.Enqueue(node.left);
            if (node.right != null)
                q2.Enqueue(node.right);
        }
        visited.Sort();
        return visited;
    }
}